core/src/edu/asu/c3simulator/screens/MarketingScreen.java

```diff
@@ -0,0 +1,368 @@
+			selectedMarketingLabel = associatedLabel;
+			marketingMethodChosen = true;
+		}
+	}
+	
+	private Game game;
+	private Stage stage;
+	private Skin skin;
+	private Product selectedProduct;
+	private Label selectedProductLabel, selectedMarketingLabel;
+	private TextButton companyPromotionButton;
+	private String selectedMarketingMethod = "";
+	private boolean promoteCompany = false, productChosen = false,
+			marketingMethodChosen = false;
+	
+	public MarketingScreen(Game game)
+	{
+		this.game = game;
+		this.stage = new Stage();
+		this.skin = new Skin(Gdx.files.internal("skins/default/uiskin.json"));
+		
+		// TODO add Corner Advisor
+		// TODO add Home Button
+		
+		NavigationPanel navigationPanel = createNavigationPanel();
+		
+		Table mainTable = createMainTable();
+		mainTable.setPosition(
+				stage.getWidth() - 0.01f * stage.getWidth() - mainTable.getWidth(),
+				0.01f * stage.getHeight());
+		
+		stage.addActor(navigationPanel);
+		stage.addActor(mainTable);
+	}
+	
+	/**
+	 * The method creates the navigation panel.
+	 * 
+	 * @return the NavigationPanel actor that was created.
+	 */
+	private NavigationPanel createNavigationPanel()
+	{
+		NavigationPanel navigationPanel = new NavigationPanel(game, skin);
+		
+		// TODO add screens
+		navigationPanel.addButton("Products", null);
+		navigationPanel.addSubButton("Products", "Pre-Market", null);
+		navigationPanel.addSubButton("Products", "Current", null);
+		navigationPanel.addSubButton("Products", "Retired", null);
+		navigationPanel.addButton("Product Growth", null);
+		navigationPanel.addSubButton("Product Growth", "Supply", null);
+		navigationPanel.addSubButton("Product Growth", "Demand", null);
+		navigationPanel.addButton("Employment", null);
+		navigationPanel.addButton("Marketing", null);
+		
+		navigationPanel.setPosition(0.01f * stage.getWidth(), stage.getHeight()
+				- (0.3f * stage.getHeight()));
+		
+		return navigationPanel;
+	}
+	
+	/**
+	 * This method creates the main section of the GUI
+	 * 
+	 * @return The table that was created.
+	 */
+	private Table createMainTable()
+	{
+		Table mainTable = new Table();
+		mainTable.setSize(0.80f * stage.getWidth(), 0.80f * stage.getHeight());
+		
+		Table leftSection = new Table();
+		Table rightSection = new Table();
+		
+		ScrollPane productGridScroll = new ScrollPane(getProductGrid(), skin);
+		companyPromotionButton = new TextButton("Promote Company", skin);
+		companyPromotionButton.addListener(new ClickListener() {
+			@Override
+			public void clicked(InputEvent event, float x, float y)
+			{
+				if (promoteCompany)
+					companyPromotionButton.setColor(1.0f, 1.0f, 1.0f, 1.0f);
+				else
+					companyPromotionButton.setColor(0.0f, 1.0f, 0.0f, 1.0f);
+				promoteCompany = !promoteCompany;
+				productChosen = false;
+				if (selectedProduct != null)
+					selectedProduct = null;
+				if (selectedProductLabel != null)
+				{
+					selectedProductLabel.setColor(1.0f, 1.0f, 1.0f, 1.0f);
+					selectedProductLabel = null;
```

Comment:
Why are we setting the color and then discarding the object? Is it referenced elsewhere?